LocalMultiplayerAgent/DeploymentTool/DeploymentScript.cs

```diff
@@ -0,0 +1,256 @@
+                    VmSize = (AzureVmSize)Enum.Parse(typeof(AzureVmSize), settingsDeployment.VmSize)
+
+                }).ToList(),
+                BuildName = settingsDeployment.BuildName,
+                GameAssetReferences = settings.AssetDetails?.Select(x => new AssetReferenceParams()
+                {
+                    FileName = GetAssetFileNameFromPath(x.LocalFilePath),
+                    MountPath = x.MountPath
+                }).ToList(),
+                StartMultiplayerServerCommand = settings.ContainerStartParameters.StartGameCommand,
+                ContainerFlavor = ContainerFlavor.ManagedWindowsServerCore
+            };
+        }
+
+        
+
+        public CreateBuildWithProcessBasedServerRequest GetProcessBasedServerRequest()
+        {
+            return new CreateBuildWithProcessBasedServerRequest
+            {
+                VmSize = (AzureVmSize)Enum.Parse(typeof(AzureVmSize), settingsDeployment.VmSize),
+                GameCertificateReferences = null,
+                Ports = PortMapping(),
+                MultiplayerServerCountPerVm = settingsDeployment.MultiplayerServerCountPerVm,
+                RegionConfigurations = settingsDeployment.RegionConfigurations?.Select(x => new BuildRegionParams()
+                {
+                    Region = x.Region,
+                    MaxServers = x.MaxServers,
+                    StandbyServers = x.StandbyServers,
+                    MultiplayerServerCountPerVm = settingsDeployment.MultiplayerServerCountPerVm,
+                    VmSize = (AzureVmSize)Enum.Parse(typeof(AzureVmSize), settingsDeployment.VmSize)
+
+                }).ToList(),
+                BuildName = settingsDeployment.BuildName,
+                GameAssetReferences = settings.AssetDetails?.Select(x => new AssetReferenceParams()
+                {
+                    FileName = GetAssetFileNameFromPath(x.LocalFilePath),
+                }).ToList(),
+                StartMultiplayerServerCommand = settings.ProcessStartParameters.StartGameCommand,
+                OsPlatform = settingsDeployment.OSPlatform
+            };
+        }
+
+        public async Task<PlayFabResult<CreateBuildWithProcessBasedServerResponse>> CreateBuildWithProcessBasedServer(CreateBuildWithProcessBasedServerRequest request)
+        {
+            Console.WriteLine($"Starting deployment {request.BuildName} for titleId, regions  {string.Join(", ", request.RegionConfigurations.Select(x => x.Region))}");
+
+            return await PlayFabMultiplayerAPI.CreateBuildWithProcessBasedServerAsync(request);
+        }
+
+        public async Task<PlayFabResult<CreateBuildWithManagedContainerResponse>> CreateBuildWithManagedContainer(CreateBuildWithManagedContainerRequest request)
+        {
+            Console.WriteLine($"Starting deployment {request.BuildName} for titleId, regions  {string.Join(", ", request.RegionConfigurations.Select(x => x.Region))}");
```

Comment:
DRY (don't repeat yourself) this out - make a quickie method that logs deployment info and takes a request as a param

Reply:
And then use it in all the places